src/schema/evaluatorSchemas.ts

```diff
@@ -1,11 +1,64 @@
-import {ObjectType, NumberType, JsonObjectType} from '../validation';
+import {
+    ObjectType,
+    NumberType,
+    JsonObjectType,
+    UnionType,
+    StringType,
+    NullType,
+    BooleanType,
+    TypeSchema,
+    ArrayType,
+} from '../validation';
+
+type PrimitiveSchemas = [NullType, NumberType, BooleanType, StringType];
+
+function createJsonSchema(maximumDepth: number): TypeSchema {
+    const primitiveSchemas: PrimitiveSchemas = [
+        new NullType(),
+        new NumberType(),
+        new BooleanType(),
+        new StringType({maxLength: 255}),
+    ];
+
+    const getNestedSchemas = (): [...PrimitiveSchemas, ...TypeSchema[]] => [
```

Comment:
This is a replica, just a little more verbose because in PHP we can type the array and the associative array at once, but here we need to type the array and the object independently